etc/gtfs/sql.d/31-geoserver-stations.sql

```diff
@@ -1,61 +1,74 @@
 CREATE MATERIALIZED VIEW geoserver.stations_with_served_routes AS
-	SELECT
+	SELECT 
 		station_id,
-		-- todo: is there something faster than min()? we just want *any*
-		min(station_name) AS station_name,
-		ST_Centroid(ST_Collect(stop_locs)) AS centroid,
-		-- prio_encoded_route_type has a multiple of 10000 added, which determines render prio (less is higher prio)
-		-- the last 4 digits represent the original route_type, so we need to modulo...
-		mod(min(prio_encoded_route_type::int),10000) AS prio_route_type,
-		json_object_agg(route_type, route_names) AS routes,
-		-- prio_encoded_route_type has a multiple of 10000 added, which determines render prio (less is higher prio)
-		min(prio_encoded_route_type::int) / 10000 AS render_prio
-	FROM (
-		-- determine available routes for each (station, route_type)
+		station_name,
+		-- prefer original parent station coord if available, else centroid of quays
+		coalesce(s.stop_loc, centroid) centroid,
```

Comment:
You could also pass the station's `stop_loc` through from the innermost `SELECT` (where a `JOIN` with `stations` already happens). But this works too, I don't mind either way.